metadata-ingestion/tests/unit/sql_parsing/test_sql_aggregator.py

```diff
@@ -1025,3 +1026,75 @@ def test_sql_aggreator_close_cleans_tmp(tmp_path):
         assert len(os.listdir(tmp_path)) > 0
         aggregator.close()
         assert len(os.listdir(tmp_path)) == 0
+
+
+@freeze_time(FROZEN_TIME)
+def test_diamond_problem(pytestconfig: pytest.Config, tmp_path: pathlib.Path) -> None:
+    aggregator = SqlParsingAggregator(
+        platform="snowflake",
+        generate_lineage=True,
+        generate_usage_statistics=False,
+        generate_operations=False,
+        is_temp_table=lambda x: x.lower()
+        in [
+            "diamond_problem.dummy_test.diamond_problem.t1",
+            "diamond_problem.dummy_test.diamond_problem.t2",
+            "diamond_problem.dummy_test.diamond_problem.t3",
+            "diamond_problem.dummy_test.diamond_problem.t4",
+        ],
+    )
+
+    aggregator._schema_resolver.add_raw_schema_info(
+        DatasetUrn(
+            "snowflake", "diamond_problem.dummy_test.diamond_problem.diamond_source1"
+        ).urn(),
+        {"col_a": "int", "col_b": "int", "col_c": "int"},
+    )
+
+    aggregator._schema_resolver.add_raw_schema_info(
+        DatasetUrn(
+            "snowflake",
+            "diamond_problem.dummy_test.diamond_problem.diamond_destination",
+        ).urn(),
+        {"col_a": "int", "col_b": "int", "col_c": "int"},
+    )
+
+    # Diamond query pattern: source1 -> t1 -> {t2, t3} -> t4 -> destination
+    queries = [
+        "CREATE TEMPORARY TABLE t1 as select * from diamond_source1;",
+        "CREATE TEMPORARY TABLE t2 as select * from t1;",
+        "CREATE TEMPORARY TABLE t3 as select * from t1;",
```

Comment:
might be useful to validate that things like column renames and multiple upstream columns for a given downstream one still work in a diamond